libs/blockscout-chains/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+use reqwest_middleware::ClientBuilder;
+use reqwest_retry::{policies::ExponentialBackoff, RetryTransientMiddleware};
+use serde::Deserialize;
+use std::collections::HashMap;
+
+const CHAINS_URL: &str = "https://chains.blockscout.com/api/chains";
+
+pub async fn get_blockscout_chains() -> anyhow::Result<BlockscoutChains> {
+    let max_retries = 3;
+    let retry_policy = ExponentialBackoff::builder().build_with_max_retries(max_retries);
+    let client = ClientBuilder::new(reqwest::Client::new())
+        .with(RetryTransientMiddleware::new_with_policy(retry_policy))
+        .build();
+    let res = client.get(CHAINS_URL).send().await?;
+    let chains: BlockscoutChains = res.json().await?;
+    Ok(chains)
+}
+
+pub type BlockscoutChains = HashMap<i64, BlockscoutChainData>;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct BlockscoutChainData {
+    pub name: String,
+    pub description: String,
+    pub ecosystem: Ecosystem,
+    pub is_testnet: Option<bool>,
+    pub layer: Option<u8>,
```

Comment:
Nit: also, when it may be missing?

Reply:
Even though the original interface specifies this field to be always present, I found some chains where it is `null`.
You can check the response of https://chains.blockscout.com/api/chains and search for `"layer": null`

Reply:
Get it, thanks. Let's maybe ask them to fix that inside their interface as well?

Reply:
Decided to leave the Option type as was discussed on the call